fclient/federationclient.go

```diff
@@ -14,9 +15,60 @@ import (
 	"golang.org/x/crypto/ed25519"
 )
 
+// an interface for gmsl.FederationClient - contains functions called by federationapi only.
+type FederationClient interface {
+	gomatrixserverlib.KeyClient
+
+	SetInternalClient(client Client)
+	WithUserAgent(agent string)
```

Comment:
I meant to use this as a `ClientOption` for use with `NewFederationClient`

Reply:
ah I see